platform/darwin/src/MGLCustomVectorSource.h

```diff
@@ -0,0 +1,65 @@
+#import "MGLSource.h"
+#import "MGLGeoJSONSource.h"
+#import "MGLGeoJSONSourceBase.h"
+#import "MGLGeometry.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol MGLFeature;
+
+/**
+ Data source for `MGLCustomVectorSource`.
+ */
+@protocol MGLCustomVectorSourceDataSource <NSObject>
+
+/** 
+ Fetch data for a tile
+ @param zoom
+ @param y
+ @param x
+ */
+- (NSArray<id <MGLFeature>>*)getTileForZoomLevel:(NSUInteger)zoomLevel x:(NSUInteger)x y:(NSUInteger)y;
```

Comment:
Now that this is a simple getter, it should be called `-featuresInTileAtX:y:zoomLevel:`, without the “get”.

I think it would be just as generally useful to have a `-featuresInCoordinateBounds:zoomLevel:`, per https://github.com/mapbox/mapbox-gl-native/pull/6940#discussion_r87365181, to accommodate those migrating from MKOverlayRenderer. If you think it’s worth keeping a tile coordinate–based API around, perhaps both methods could be optional methods on this protocol. MGLCustomVectorSource would end up calling one or the other, depending on which one is implemented.